### ISSUE TITLE
[BottomSheetDialogFragment] Top inset impossible to remove

### ISSUE THREAD

**Issue by ParticleCore** [material-components/material-components-android#3389]
**Description:** I am currently trying to create a fullscreen/full height `BottomSheetDialogFragment`, but since it has a very weird behavior implemented where it adds and retracts insets while the sheet is sliding in and out, I decided to disable any inset handling done by the component and manually take care of the insets. However, I found a problem with that approach as well. No matter what I try, the component still forces a top inset that is the same size of the status bar even when disabling any and all inset handling flags I could find checked in the component source code, ex.:

https://github.com/material-components/material-components-android/blob/46778db594edc5915987dcd6e737fa7d15414690/lib/java/com/google/android/material/bottomsheet/BottomSheetBehavior.java#L1726-L1734

The objective is for the component screen to fill the entire screen of the device, but as it currently is the top area of the status bar can never be covered, meaning the component will never display underneath the status bar as desired.

**Expected behavior:** Screenshots are attached,
 - Screen 1 is meant to launch the component
![screen1 (25%)](https://github.com/material-components/material-components-android/assets/9222661/2d40a050-1d73-4efb-bcdc-6220b8495a76)

 - Screen 2 is what is currently happening, notice the top area that cannot be covered
![screen2 (25%)](https://github.com/material-components/material-components-android/assets/9222661/d1cf466d-9103-428d-bc74-2805a24b9484)

 - Screen 3 is what it should be looking like at the top (I created a separat activity just to represent how screen 2 should be looking)
![screen3 (25%)](https://github.com/material-components/material-components-android/assets/9222661/183e9f5c-a08c-495a-8017-460a0930338f)

**Source code:** Not applicable.

**Minimal sample app repro:** Sample follows attached.
[BottomSheetInsetBug.zip](https://github.com/material-components/material-components-android/files/11437611/BottomSheetInsetBug.zip)

**Android API version:** Any android version since 24 (7.0 Nougat) till most recent 33 (13.0). Tested all versions and every single one of them have the same issue.

**Material Library version:** Material Android Library version `com.google.android.material:material:1.8.0`

**Device:** Any real device and emulator tested.

**Comment by imhappi** [comment]
I don't believe we want to remove the top inset for the bottom sheet, as content could get cut off eg. on devices with front cameras overlaying the screen

However, you should be able to achieve getting the bottom sheet under a transparent status bar, which would make it the same color as the bottom sheet.

See this example of a full-screen bottom sheet that goes under the status bar can be found on our test app:
https://github.com/material-components/material-components-android/blob/master/catalog/java/io/material/catalog/bottomsheet/BottomSheetMainDemoFragment.java

**Comment by ParticleCore** [comment]
@imhappi 

Please reopen, that is not the same thing as I was pointing out. I did explain that I was trying to achieve this because the current implementation of the component is adding the insets in a dynamic way which causes any content inside of it to squeeze and stretch as the user slides the component up or down. As that happens, the top and bottom insets are being incremented/decremented as the component slides vertically.

That is the main reason why I wanted to just turn off that entire behavior completely and handle it manually, but even when I do that the component is permanently adding a top inset which cannot be disabled by any available option.

**Comment by ParticleCore** [comment]
@imhappi thanks for reopening, I wasn't sure if this repo was set to ignore closed cases or not.

**Comment by oOJohn6Oo** [comment]
I think you are not using it in a right way, try the style below.

```
<style name="MyModalBottomSheetDialog" parent="@style/Theme.MaterialComponents.BottomSheetDialog">
    <item name="enableEdgeToEdge">true</item>
    <item name="android:windowIsFloating">false</item>
    <item name="paddingBottomSystemWindowInsets">false</item>
    <item name="paddingTopSystemWindowInsets">false</item>
</style>
```

**Comment by oOJohn6Oo** [comment]
> I think you are not using it in a right way, try the style below.
> 
> ```
> <style name="MyModalBottomSheetDialog" parent="@style/Theme.MaterialComponents.BottomSheetDialog">
>     <item name="enableEdgeToEdge">true</item>
>     <item name="android:windowIsFloating">false</item>
>     <item name="paddingBottomSystemWindowInsets">false</item>
>     <item name="paddingTopSystemWindowInsets">false</item>
> </style>
> ```

Oh sorry, it's not work. That's truly a bug. But I do have a way to prevent BottomSheet's weird behavior.

1. Override DialogFragment's `onViewCreated` function.
2. Set a WindowInsetsListener to android.R.id.content, and consume all insets.
``` kotlin
ViewCompat.setOnApplyWindowInsetsListener(desiredView) { _, insets ->
    // customize your view here

    WindowInsetsCompat.CONSUMED
}
```

**Comment by ParticleCore** [comment]
> > I think you are not using it in a right way, try the style below.
> > ```
> > <style name="MyModalBottomSheetDialog" parent="@style/Theme.MaterialComponents.BottomSheetDialog">
> >     <item name="enableEdgeToEdge">true</item>
> >     <item name="android:windowIsFloating">false</item>
> >     <item name="paddingBottomSystemWindowInsets">false</item>
> >     <item name="paddingTopSystemWindowInsets">false</item>
> > </style>
> > ```
> 
> Oh sorry, it's not work. That's truly a bug. But I do have a way to prevent BottomSheet's weird behavior.
> 
>     1. Override DialogFragment's `onViewCreated` function.
> 
>     2. Set a WindowInsetsListener to android.R.id.content, and consume all insets.
> 
> 
> ```kotlin
> ViewCompat.setOnApplyWindowInsetsListener(desiredView) { _, insets ->
>     // customize your view here
> 
>     WindowInsetsCompat.CONSUMED
> }
> ```

Thanks for the suggestion, but then the rest of the app will no longer receive the insets, and to work around that you have to create your own inset manager and it becomes a complete nightmare. So far the best compromise I found was to make the height a static value by calculating exactly how tall it should be when fully opened, but this might break in the future I believe.

**Comment by bharat1320** [comment]
Guys this worked like a charm for me,

`        // Set dialog to full-screen, removing the top insets for notification bar
        dialog?.window?.apply {
            setFlags(
                WindowManager.LayoutParams.FLAG_LAYOUT_NO_LIMITS,
                WindowManager.LayoutParams.FLAG_LAYOUT_NO_LIMITS
            )
        }`

**Comment by bharat1320** [comment]
`
        // Set dialog to full-screen, removing the top insets for notification bar
        dialog?.window?.apply {
            setFlags(
                WindowManager.LayoutParams.FLAG_LAYOUT_NO_LIMITS,
                WindowManager.LayoutParams.FLAG_LAYOUT_NO_LIMITS
            )
        }
`

**Comment by bharat1320** [comment]
`
ViewCompat.setOnApplyWindowInsetsListener(desiredView) { _, insets ->
    // customize your view here

    WindowInsetsCompat.CONSUMED
}
`

**Comment by wman1980** [comment]
Thanks @bharat1320! It was driving me (almost) crazy ...

Don't know if this is a bug or its on purpose, but to me I would expect the BottomSheetDialogFragment to draw behind statusbar when edgeToEdge is enabled and than the instance of the BottomSheetDialogFragment should handle the top insets.

**Comment by manabu-nakamura** [comment]
3 years ago, I also used ViewCompat.setOnApplyWindowInsetsListener().
Now, I add a BottomSheetDragHandleView to the BottomSheet and don't use ViewCompat.setOnApplyWindowInsetsListener().
It makes the corners of the BottomSheet rounded.
https://m3.material.io/components/bottom-sheets/overview:
>Shape: Bottom sheets have a 28dp top corner radius

#2969
See also #3173.
